src/main/java/org/stellar/sdk/Sep10Challenge.java

```diff
@@ -40,4 +46,366 @@ public static String newChallenge(
 
     return transaction.toEnvelopeXdrBase64();
   }
+
+  /**
+   * Returns a valid <a href="https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#response" target="_blank">SEP 10</a> challenge, for use in web authentication.
+   *
+   * @param signer          The server's signing account.
+   * @param network         The Stellar network used by the server.
+   * @param clientAccountId The stellar account belonging to the client.
+   * @param anchorName      The name of the anchor which will be included in the ManageData operation.
+   * @param timeout         Challenge duration in seconds.
+   * @return a valid <a href="https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#response" target="_blank">SEP 10</a> challenge.
+   */
+  public static String newChallenge(
+          KeyPair signer,
+          Network network,
+          String clientAccountId,
+          String anchorName,
+          long timeout
+  ) {
+    long now = System.currentTimeMillis() / 1000L;
+    long end = now + timeout;
+    TimeBounds timeBounds = new TimeBounds(now, end);
+    return newChallenge(signer, network, clientAccountId, anchorName, timeBounds);
+  }
+
+  /**
+   * Reads a SEP 10 challenge transaction and returns the decoded transaction envelope and client account ID contained within.
+   * <p>
+   * It also verifies that transaction is signed by the server.
+   * <p>
+   * It does not verify that the transaction has been signed by the client or
+   * that any signatures other than the servers on the transaction are valid. Use
+   * one of the following functions to completely verify the transaction:
+   * {@link Sep10Challenge#verifyChallengeTransactionSigners(String, String, Network, List)} or
+   * {@link Sep10Challenge#verifyChallengeTransactionThreshold(String, String, Network, int, List)}
+   *
+   * @param challengeTransaction SEP-0010 transaction challenge transaction in base64.
+   * @param serverAccountId      Account ID for server's account.
+   * @param network              The network to connect to for verifying and retrieving.
+   * @return {@link ChallengeTransaction}, the decoded transaction envelope and client account ID contained within.
+   * @throws InvalidSep10ChallengeException If the SEP-0010 validation fails, the exception will be thrown.
+   * @throws IOException                    If read XDR string fails, the exception will be thrown.
+   */
+  public static ChallengeTransaction readChallengeTransaction(String challengeTransaction, String serverAccountId, Network network) throws InvalidSep10ChallengeException, IOException {
```

Comment:
given that this function is taking an xdr representation of a transaction and parsing it into a `ChallengeTransaction` instance, I prefer the name `parseChallengeTransaction()`

Reply:
I think @overcat was keeping this similar and 1-1 with the Go and Python SDKs that already use the term `read`. Do you see any big issues continuing forward and using read here too?

Reply:
@leighmcculloch ok in that case we can keep the function name as `readChallengeTransaction `